<launch>

    <arg name="namespace" default="sam"/>

    <!--
    <node name="pid_setpoint_static" pkg="sam_loop_test" type="pid_setpoint_static" ns="LCG_trim">
      <param name="setpoint_topic" value="/pitch_setpoint" />
      <param name="setpoint" value="0.15" />
    </node>

    <node name="pid_setpoint_static" pkg="sam_loop_test" type="pid_setpoint_static" ns="VBS_depth">
      <param name="setpoint_topic" value="/depth_setpoint" />
      <param name="setpoint" value="0.5" />
    </node>


    <node name="pid_tf_listener" pkg="sam_loop_test" type="pid_tf_listener" output="screen" >
	    <param name="odom_frame" value="$(arg namespace)_odom"/>
	    <param name="base_frame" value="$(arg namespace)/base_link"/>
	    <param name="world_frame" value="$(arg namespace)_odom"/>
    </node>
-->
    <node name="pid_actuator" pkg="sam_basic_controllers" type="pid_actuator" ns="LCG_trim" >
      <param name="topic_from_controller" value="lcg_control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="1" />
      <param name="loop_frequency" value="50" />
      <remap from="/actuator_input" to="/uavcan_lcg_command"/>
    </node>

    <node name="pid_actuator" pkg="sam_basic_controllers" type="pid_actuator" ns="VBS_depth" >
      <param name="topic_from_controller" value="vbs_control_action" />
      <param name="topic_to_actuator" value="/actuator_input" />
      <param name="limit_between_setpoints" value="1" />
      <param name="loop_frequency" value="5" />
      <remap from="/actuator_input" to="/uavcan_vbs_command"/>
    </node>

    <node name="controller" pkg="pid" type="controller" ns="LCG_trim" output="screen" >
      <param name="Kp" value="160.0" />
      <param name="Ki" value="1.0" />
      <param name="Kd" value="5.0" />
      <param name="upper_limit" value="50.0" />
      <param name="lower_limit" value="-50.0" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="1000.0" />
      <param name="min_loop_frequency" value="10.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <param name="topic_from_plant" value="/pitch_feedback" />
      <param name="topic_from_controller" value="lcg_control_action" />
      <remap from="setpoint" to="/pitch_setpoint" />
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     </node>

    <node name="controller" pkg="pid" type="controller" ns="VBS_depth" output="screen" >
      <param name="Kp" value="150.0" />
      <param name="Ki" value="0" />
      <param name="Kd" value="0" />
      <param name="upper_limit" value="50.0" />
      <param name="lower_limit" value="-50.0" />
      <param name="windup_limit" value="100" />
      <param name="max_loop_frequency" value="1000.0" />
      <param name="min_loop_frequency" value="10.0" />
      <!--CHANGE LAUNCH FILE TO CHOOSE SUBSCRIBER AND PUBLISHER TOPICS!! /-->
      <param name="topic_from_plant" value="/depth_feedback" />
      <param name="topic_from_controller" value="vbs_control_action" />
      <remap from="setpoint" to="/depth_setpoint" />
      <!--remap from="/plant_state_pitch" to="state" />
      <remap from="control_effort" to="/control_effort" /-->
     </node>

</launch>
